src/Symfony/Component/TypeInfo/Type/NullableType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\Type;
+
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\TypeIdentifier;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @template T of Type
+ *
+ * @extends UnionType<T|BuiltinType<TypeIdentifier::NULL>>
+ * @implements WrappingTypeInterface<T>
+ *
+ * @experimental
+ */
+final class NullableType extends UnionType implements WrappingTypeInterface
+{
+    /**
+     * @param T $type
+     */
+    public function __construct(
+        private readonly Type $type,
+    ) {
+        if ($type instanceof UnionType) {
+            parent::__construct(Type::null(), ...$type->getTypes());
+
+            return;
+        }
+
+        parent::__construct(Type::null(), $type);
+    }
+
+    public function getWrappedType(): Type
```

Comment:
this type actually composes both this type and the NULL type. It might cause confusion for code not handling properly types implementing both WrappingTypeInterface and IntersectionType.

And this `getWrappedType` is not even a replacement for `asNonNullable` as nothing prevents the NULL type to be part of the wrapped type.

Reply:
> this type actually composes both this type and the NULL type. It might cause confusion for code not handling properly types implementing both WrappingTypeInterface and IntersectionType.

I'm not sure to understand what you mean here, could you elaborate please?

> And this `getWrappedType` is not even a replacement for `asNonNullable` as nothing prevents the NULL type to be part of the wrapped type.

From now on, the `NullableType` can only be constructed with a non-nullable type, therefore `asNonNullable` is now equal to `getWrappedType`.